src/cli.js

```diff
@@ -128,18 +139,11 @@ if (cli.flags.quiet) {
 }
 
 if (cli.flags.syntax) {
-  optionsBase.syntax = cli.flags.syntax
+  optionsBase.syntax = resolveModule(cli.flags.syntax)
```

Comment:
Since the built-in syntax options are dependencies, it should resolve to the same dependency. If that doesn't sound like a good plan, we can either put a guard here for the built-in syntaxes or use the extracted module resolution function to resolve the module in `getPostcssResult` instead (my preference).